neo/Prompt/Commands/Show.py

```diff
@@ -0,0 +1,159 @@
+            return
+
+        try:
+            return self.execute_sub_command(item, arguments[1:])
+        except KeyError:
+            print(f"{item} is an invalid parameter")
+            return
+
+
+class CommandShowBlock(CommandBase):
+    def __init__(self):
+        super().__init__()
+
+    def execute(self, arguments):
+        item = get_arg(arguments)
+        txarg = get_arg(arguments, 1)
+
+        block = Blockchain.Default().GetBlock(item)
+
+        if block is not None:
+            block.LoadTransactions()
+
+            if txarg and 'tx' in txarg:
+                txs = []
+                for tx in block.FullTransactions:
+                    print(json.dumps(tx.ToJson(), indent=4))
+                    txs.append(tx.ToJson())
+                return txs
+
+            print(json.dumps(block.ToJson(), indent=4))
+            return block.ToJson()
+
+        else:
+            print("Could not locate block %s" % item)
+            return
+
+    def command_desc(self):
+        p1 = ParameterDesc('index/hash', 'the index or scripthash of the block')
+        p2 = ParameterDesc('tx', 'arg to only show block transactions', optional=True)
+        return CommandDesc('block', 'show a specified block', [p1, p2])
+
+
+class CommandShowHeader(CommandBase):
+    def __init__(self):
+        super().__init__()
+
+    def execute(self, arguments):
+        item = get_arg(arguments)
+
+        header = Blockchain.Default().GetHeaderBy(item)
+        if header is not None:
+            print(json.dumps(header.ToJson(), indent=4))
+            return header.ToJson()
+        else:
+            print("Could not locate header %s\n" % item)
+            return
+
+    def command_desc(self):
+        p1 = ParameterDesc('index/hash', 'the index or scripthash of the block header')
+        return CommandDesc('header', 'show the header of a specified block', [p1])
+
+
+class CommandShowTx(CommandBase):
+    def __init__(self):
+        super().__init__()
+
+    def execute(self, arguments):
+        try:
+            txid = UInt256.ParseString(get_arg(arguments))
+            tx, height = Blockchain.Default().GetTransaction(txid)
+            if height > -1:
+                jsn = tx.ToJson()
+                jsn['height'] = height
+                jsn['unspents'] = [uns.ToJson(tx.outputs.index(uns)) for uns in
+                                   Blockchain.Default().GetAllUnspent(txid)]
+                print(json.dumps(jsn, indent=4))
+                return jsn
+            else:
+                print(f"Could not find transaction for hash {txid}")
+                return
+        except Exception as e:
+            print("Could not find transaction from args: %s (%s)" % (e, arguments))
+            return
+
+    def command_desc(self):
+        p1 = ParameterDesc('hash', 'the scripthash of the transaction')
+        return CommandDesc('tx', 'show a specified transaction', [p1])
```

Comment:
Should this maybe be? (Testing my English grammar here)
`show _the_ specified transaction` or `show a _specific_ transaction`

Reply:
I think those are all correct. What is your preference? (What sounds the best to you out loud?)

Reply:
If you say they're all good, then leave it as is. I'm not the native speaker here. It just felt odd when reading, but it might just be me not being used to that form.